src/clj_gatling/simulation.clj

```diff
@@ -16,17 +16,23 @@
 (defn- now [] (System/currentTimeMillis))
 
 (defn asynchronize [f ctx]
-  (let [parse-response (fn [result]
-                         (if (vector? result)
-                           {:result (first result) :end-time (now) :context (second result)}
-                           {:result result :end-time (now) :context ctx}))]
+  (let [parse-result   (fn [result]
+                         (if (instance? Throwable result)
+                           {:result false :exception result}
+                           {:result result}))
+        parse-response (fn [response]
+                         (if (vector? response)
+                           (assoc (parse-result (first response)) :end-time (now) :context (second response))
+                           (assoc (parse-result response) :end-time (now) :context ctx)))]
     (go
       (try
-        (let [result (f ctx)]
-          (if (instance? clojure.core.async.impl.channels.ManyToManyChannel result)
-            (parse-response (<! result))
-            (parse-response result)))
+        (let [response (f ctx)]
+          (if (instance? clojure.core.async.impl.channels.ManyToManyChannel response)
+            (parse-response (<! response))
+            (parse-response response)))
         (catch Exception e
+          {:result false :end-time (now) :context ctx :exception e})
+        (catch AssertionError e
           {:result false :end-time (now) :context ctx :exception e})))))
```

Comment:
We catch `AssertionError` so that users can easily make use of `(assert ...)` in their steps, for validation, but `AssertionError` derives directly from `Throwable`, not `Exception`, and we don't want to catch all `Throwable`s, so we don't catch more critical issues.

Reply:
I think this is a non backwards compatible change. I suggest that this change is not part of this PR. This can be an separate issue (e.g. catch also AssertionErrors) where we can have a discussion about this.

Reply:
Hmm, OK. I'm not sure why you'd _not_ want to catch `AssertionError`. Previously it would have caused the entire test run to bomb out, so my feeling is that people can't be using `assert`. Or, if they are, are catching it and casting it to a different exception type that is handled before throwing back to `clj-gatling`.

So I guess this is _technically_ a non-backwards-compatible change, I just can't see how anyone could have been usefully using `assert` the old way. Can pull it out to a separate PR though.

Reply:
No need to apologise! It's your project/repo ;) And if it makes it clearer for others what is going on, in case they want 0.17.1 and not 0.17.2, then great!